tf2_ros/include/tf2_ros/message_filter.h

```diff
@@ -361,51 +314,53 @@ class MessageFilter : public MessageFilterBase, public message_filters::SimpleFi
 
 
     // We can transform already
-    if (info.success_count == expected_success_count_)
-    {
+    if (info.success_count == expected_success_count_) {
       messageReady(evt);
-    }
-    else
-    {
+    } else {
       // If this message is about to push us past our queue size, erase the oldest message
-      if (queue_size_ != 0 && message_count_ + 1 > queue_size_)
-      {
+      if (queue_size_ != 0 && message_count_ + 1 > queue_size_) {
+
         // While we're using the reference keep a shared lock on the messages.
-        boost::shared_lock< boost::shared_mutex > shared_lock(messages_mutex_);
+        std::lock_guard<std::mutex> guard_lock(messages_mutex_);
 
         ++dropped_message_count_;
-        const MessageInfo& front = messages_.front();
-        TF2_ROS_MESSAGEFILTER_DEBUG("Removed oldest message because buffer is full, count now %d (frame_id=%s, stamp=%f)", message_count_,
-                                (mt::FrameId<M>::value(*front.event.getMessage())).c_str(), mt::TimeStamp<M>::value(*front.event.getMessage()).toSec());
+        const MessageInfo & front = messages_.front();
+        TF2_ROS_MESSAGEFILTER_DEBUG(
+          "Removed oldest message because buffer is full, count now %d (frame_id=%s, stamp=%f)",
+          message_count_,
+          (mt::FrameId<M>::value(*front.event.getMessage())).c_str(),
+          mt::TimeStamp<M>::value(*front.event.getMessage()).seconds());
 
         V_TransformableRequestHandle::const_iterator it = front.handles.begin();
         V_TransformableRequestHandle::const_iterator end = front.handles.end();
-        for (; it != end; ++it)
-        {
+        for (; it != end; ++it) {
           bc_.cancelTransformableRequest(*it);
         }
 
         messageDropped(front.event, filter_failure_reasons::Unknown);
+
         // Unlock the shared lock and get a unique lock. Upgradeable lock is used in transformable.
         // There can only be one upgrade lock. It's important the cancelTransformableRequest not deadlock with transformable.
         // They both require the transformable_requests_mutex_ in BufferCore.
-        shared_lock.unlock();
```

Comment:
Based on the previous conversation and looking at this again, I agree with @gaoethan that we don't need this anymore.  In that case, though, it's confusing to leave this in place since it doesn't apply anymore.  It is also confusing for the `if` block to take the lock at line 324, drop it (because of scope) at 351, and then take it back again at 357.  In my opinion, it would be clearer if we just unconditionally took the lock when entering the `else` (around line 320), and then let the scope unlock it at line 360.  Does that make sense?